src/app/components/ConfirmOrCancel/index.tsx

```diff
@@ -1,48 +1,53 @@
 import { MouseEventHandler } from "react";
+import { useTranslation } from "react-i18next";
 
 import Button from "../Button";
 
-type Props = {
+export type Props = {
   disabled?: boolean;
   loading?: boolean;
   label?: string;
   onConfirm: MouseEventHandler;
   onCancel: MouseEventHandler;
+  onBlock?: MouseEventHandler;
 };
 
-function ConfirmOrCancel({
+export default function ConfirmOrCancel({
   disabled = false,
   loading = false,
   label = "Confirm",
+  onBlock,
   onConfirm,
   onCancel,
 }: Props) {
+  const { t } = useTranslation("components", {
+    keyPrefix: "confirmOrCancel",
+  });
   return (
-    <div className="text-center">
-      <div className="mb-4">
+    <div className="text-center p-2 mb-2">
+      <div className="flex flex-row mb-4">
+        <Button onClick={onCancel} label={"Cancel"} halfWidth />
         <Button
           onClick={onConfirm}
           label={label}
-          fullWidth
           primary
           disabled={disabled}
           loading={loading}
+          halfWidth
         />
       </div>
 
-      <p className="mb-2 text-sm text-gray-400">
-        <em>Only connect with sites you trust.</em>
-      </p>
-
-      <a
-        className="underline text-sm text-gray-600 dark:text-neutral-400"
-        href="#"
-        onClick={onCancel}
-      >
-        Cancel
-      </a>
+      {onBlock ? (
```

Comment:
I think the block section we only need on the enable screen.
thus we probably don't need that in here, do we?

Reply:
Yes, but the catch is that the link should be placed inside the component...
Can passing it as a child work?

Reply:
what do you mean with inside the component? 

What I mean is that we do not have it inside the component, but just add it to the enable screen - or wherever it is needed

Reply:
But currently, we are using flex and justify-between to push the ConfirmOrCancel screen to the bottom, so we would have to have the block link inside it, otherwise, the workaround would be to use a prop to disable showing the usual "only_trusted" string and then add it separately and wrap the component and block link in a div. Should I do this, then?

![Screenshot from 2022-08-02 16-33-11 (copy)](https://user-images.githubusercontent.com/64399555/182438602-027b94bc-1879-4e0f-b5cd-9320dd1dbda9.png)

Reply:
no I would not add new props. Rather less props / less options and less code is always better (less complexity that needs to be managed) 

@escapedcat do you have an idea? 

I would actually like to have those text out of the ConfirmOrCancel component and also have the "block" option only on the enable screen. So I don't think the ConfirmOrCancel should know about the blocking